### PR TITLE
Use new colors for state icon

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Only `climate`, `alarm_control_panel`and `plant` had colored icon. Others domain was yellow `#fdd835` when active.
These PR brings the same color used for state history(https://github.com/home-assistant/frontend/pull/14330) and tile card.

`alarm_control_panel`, `binary_sensor`, `cover`, `climate`, `lock`, `light`, `humidifier`, `media_player`, `person`, `device_tracker`, `sensor`, `vacuum`, `sun`, and `update` domain now have color based on state. Other domain can be colored with future PR.

For other domain, the active color remains yellow `#fdd835` like before.
If the entity is inactive, the color is the same as before.

#### Before

image

#### After
image

#### Before
Capture d’écran 2022-11-10 à 20 03 51

#### After
Capture d’écran 2022-11-10 à 20 03 33

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by alex3305** [comment]
Looks great @piitaya, but isn't the before/after the other way around?

**Comment by piitaya** [comment]
> Looks great @piitaya, but isn't the before/after the other way around?

Yes I edited the description 😅

**Comment by Mariusthvdb** [comment]
wow, this is really nice. Finally some more intuitive coloring in the core Frontend. Please let me ask, if we can set the color scheming in  themes. Eg, I have asked before to be able to set alert_color. You now seem to implement that, only hard-coded to red/green?

Given the fact many users have carefully curated themes, it would be wonderful to be able to set those colors in a theme too?

**Comment by piitaya** [comment]
Yes, these colors can be edited with themes.

Here are the rules : 
- **active state** have a **color** based of state and domain (e.g. some binary sensor are green or red depending of the `device_class`, temperature sensor are red, humidity one purple, etc...).
- **inactive state** have the **disabled color** (grey). `#bdbdbd` in light theme and `#6f6f6f` in dark theme so that they are less visible than the active entities on the interface.

For the `icon_color`, there was already a discussion for this (https://github.com/home-assistant/core/pull/27455). 
We want to propose a logic color palette by default. That's why we focus a lot on `device_class` so user have a the best icon/colors by default and do not have to use the customize section or the `icon` attributes for each entity.

**Comment by Mariusthvdb** [comment]
great, looking forward to ditching many custom-ui templates!

the linked discussion is very old though, and might need some revisiting. Not all entities  will be captured in a device_class, and being able to set customized  colors nonetheless would be very much appreciated. 
For that, we would need the re-installment of the icon_color attribute?

File: src/components/entity/state-badge.ts [hunk 1]
```diff
@@ -100,11 +107,15 @@ export class StateBadge extends LitElement {
...
         hostStyle.backgroundImage = `url(${imageUrl})`;
         this._showIcon = false;
-      } else if (stateObj.state === "on") {
-        if (this.stateColor !== false && stateObj.attributes.rgb_color) {
+      } else if (stateActive(stateObj) && this._stateColor) {
+        const iconColor = stateColor(stateObj);
+        if (iconColor) {
+          iconStyle.color = `rgb(var(--rgb-state-${iconColor}-color))`;
+        }
+        if (stateObj.attributes.rgb_color) {
```
Comment (bramkragten) [inline src/components/entity/state-badge.ts:115]: Let's flip these and do an `else if`

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

**Comment by SmarterHomeLife** [comment]
Previously I used --paper-item-icon-active-color: var(--white) in card-mod style to specify the active color of a state-icon in picture-elements cards for camera feeds, since the change I've amended those to use --rgb-state-binary-sensor-color but this doesn't work if I refer to var(--white), it only works correctly if I use the RGB code 255,255,255.

Do theme variables no longer work on the new state icons?

**Comment by jouster1974** [comment]
this looks like a great idea.

When I'm using Chips in Mushroom, the Armed Home icon is green...but so is the Armed Away one...and my custom alarm modes, , night and garden are both grey/empty shields....it does of course say in text what mode is but just colour would be ideal and then i can do away with the text altogether.

If we could change the colours it would be great and I could tie it in with my custom icons on other dashboards, Armed away, red; Armed Home, green, so no change; Night, blue and Garden, yellow

Unless of course I have completely misinterpreted this request...either way....thanks in advance and thanks for all you do

**Pull request merged**